wicker/core/storage.py

```diff
@@ -52,36 +75,41 @@ def __init__(self) -> None:
 
-    @retry(
-        Exception,
-        tries=get_config().storage_download_config.retries,
-        backoff=get_config().storage_download_config.retry_backoff,
-        delay=get_config().storage_download_config.retry_delay_s,
-        logger=logger,
-    )
-    def download_with_retries(self, input_path: str, local_path: str):
-        try:
-            shutil.copyfile(input_path, local_path)
-        except Exception as e:
-            logging.error(f"Failed to download/move object for file path: {input_path}")
-            raise e
-
     def fetch_file(self, input_path: str, local_prefix: str, timeout_seconds: int = 120) -> str:
-        """Fetches a file system, ie: gets the path to the file.
+        """Fetch file from data storage into the local path.
 
-        This function assumes the input_path is a valid file in the file system based on wicker assumed pathing.
+        :param input_path: input file path
+        :param local_prefix: local path that specifies where to download the file
+        :return: local path to the downloaded file
+        """
+        # If we not copying the file to a new path, just return the input path.
+        if local_prefix == "":
+            return input_path
+        local_path = os.path.join(local_prefix, os.path.basename(input_path))
+        shutil.copyfile(input_path, local_path)
+        return local_path
 
-        The reasoning here is that if you use this for a mounted file system that doesn't have caching
-        to local automatically you can grab and move files to local file system on instance.
+    def put_file(self, input_path: str, target_path: str) -> None:
+        """Put file on local or mounted data storage.
 
-        :param input_path: input file path on system
-        :param local_prefix: local path that specifies where to download the file
-        :param timeout_seconds: number of seconds till timing out on waiting for the file to be downloaded
-        :return: local path to the file on the local file system
+        :param input_path: file path on the local system.
+        :param target_path: file path on local system or mounted drive.
         """
-        input_path = os.path.join(input_path, os.path.basename(local_prefix))
-        self.download_with_retries(input_path, local_prefix)
-        return local_prefix
+        # If we are not putting the file to a new location, just return.
+        if input_path == target_path:
+            return
+        os.makedirs(Path(target_path).parent, exist_ok=True)
+        shutil.copy2(input_path, target_path)
+
+    def put_object(self, object_bytes: bytes, target_path: str) -> None:
+        """Put object on data storage in target location.
+
+        :param object_bytes: bytes to write to path
+        :param target_path: path to write object
+        """
+        os.makedirs(Path(target_path).parent, exist_ok=True)
+        with open(target_path, "wb") as binary_file:
+            binary_file.write(object_bytes)
 
 
```

Comment:
Below in the `S3DataStorage` class, we'll make a backwards compatible change so that `put_file_s3` and `put_object_s3` functions are changed to call `put_file` and `put_object` instead (which do the exactly the same thing they did before).